### ISSUE TITLE
zoom by rectangule tool doesn't work properly

### ISSUE THREAD

**Issue by josemvm** [3liz/lizmap-web-client#319]
hi!

problem detected after last commits

**Comment by rldhont** [comment]
Can you be more explicit ?
I didn't see some disfunctionnality.

**Comment by josemvm** [comment]
when i draw a small or a tiny rectangule over the map (around a point) the zoom result doesn´t match to the selected area. the result corresponds to a larger area.

**Comment by rldhont** [comment]
Do you go to the min scale ? or can you continue to zoom after your zoom by rectangle ?

**Comment by josemvm** [comment]
no, i don' t go to minscale, i can continue to make zoom after zoom by rectangle.

**Comment by rldhont** [comment]
Can you provide an example ? I can't reproduce your issue.

**Comment by josemvm** [comment]
yes, you can try at http://pdm-revisao.municipioarraiolos.pt

user: test
pass: test

thanks for your attention

**Comment by rldhont** [comment]
Hi @josemvm 
It's not an issue but the way the OpenLayers Box Handler works.
```
    endBox: function(end) {
        var result;
        if (Math.abs(this.dragHandler.start.x - end.x) > 5 ||    
            Math.abs(this.dragHandler.start.y - end.y) > 5) {   
            var start = this.dragHandler.start;
            var top = Math.min(start.y, end.y);
            var bottom = Math.max(start.y, end.y);
            var left = Math.min(start.x, end.x);
            var right = Math.max(start.x, end.x);
            result = new OpenLayers.Bounds(left, bottom, right, top);
        } else {
            result = this.dragHandler.start.clone(); // i.e. OL.Pixel
        } 
        this.removeBox();

        this.callback("done", [result]);
    }
```
If the rectangle has a width or height less than 5 pixel, the handler returns a point.

If the ZoomBox Control receives a point, the map is zoomed by one zoom level.

I close this issue because it's not a lizmap issue. Thanks for your feedback.

**Comment by josemvm** [comment]
thanks and sorry @rldhont, 

but same commits ago, this work's properly in the same projects (include montpellier and montpellier_intranet).

please, can you try one more time this version (at commit 7aea58e on May 03, 2016)
http://pdm-revisao.municipioarraiolos.pt
user: test
pass: test

is it new in the OpenLayers or is it due to the last commits?

**Comment by rldhont** [comment]
Thanks for this revert.
I prefer the effect in last commits but I can't really reproduce your issue. I'm not sure to do what you do.

**Comment by josemvm** [comment]
in my opinion the zoom by rectangule tool loses some functionality, because you should be able to get the minimum scale (maximum zoom) by drawing a small rectangule over the map as it used happened.

**Comment by rldhont** [comment]
We didn't change the ZoomBox Control.
I'm sorry to not be able to view a difference between now and before in the way to zoom to object.

We made 3 changes in map.js since the 3rd of may:
https://github.com/3liz/lizmap-web-client/commit/6e320800f5377027b9ddf81d5414fb0cfcc2af4f#diff-d977fdbe119bcd4fd2541c4c0c6ac016
https://github.com/3liz/lizmap-web-client/commit/60fc25cd92357deb5323a73d366fc18519a35c5f#diff-d977fdbe119bcd4fd2541c4c0c6ac016
https://github.com/3liz/lizmap-web-client/commit/f56a02071ec68350f853dfa862ef40a8490345d1#diff-d977fdbe119bcd4fd2541c4c0c6ac016

So do you request an enhancement about the ZoomBox ?

**Comment by josemvm** [comment]
i will try to show you ASAP. ok?

thanks

**Comment by rldhont** [comment]
Hi @josemvm in the video you provide to me, the baselayer is Google Maps.
Does the issue persist without any Google base layers ?

(I can't reproduce your issue on Montpellier or Montpellier Intranet map)

**Comment by josemvm** [comment]
hi @rldhont,

yes the issue persist without any Google base layers, also with a empty baselayer

**Comment by rldhont** [comment]
Hi @josemvm, 
In which Web browser do you have this issue ?
This is a significant question I forgot to ask you!

**Comment by josemvm** [comment]
hi @rldhont,

i'm running tests on firefox, chrome and opera

**Comment by rldhont** [comment]
And you have the same issue with this 3 different web browsers ?

**Comment by josemvm** [comment]
yes, precisely!

**Comment by josemvm** [comment]
@rldhont,

you can reproduce this issue on Lizmap Demo Intranet map by drawing a small rectangule around a special tree.

so, you can´t reach the minimum scale (maximum zoom) as previously.

**Comment by rldhont** [comment]
@josemvm I reproduce it if I draw a box around the symbol, but if my rectangle is smaller than the symbol lizmap zooms enough.

**Comment by rldhont** [comment]
This issue is linked to https://github.com/3liz/lizmap-web-client/issues/316

So If I want to fix it, I have to doesn't unfix #316

**Comment by josemvm** [comment]
hi @rldhont 

i can't reproduce that issue https://github.com/3liz/lizmap-web-client/issues/316

however i think that's more important to have the "zoom by rectangule" tool working properly, because it's a basic map functionality.

i fixed this issue for me.

if it's ok for you, you can close this issue.
